peridot/cmd/v1/peridot/task_logs.go

```
@@ -0,0 +1,190 @@
+	attrs           int
+)
+
+func init() {
+	taskLogs.Flags().StringVarP(&architecture, "architecture", "a", "", "(inop) filter by architecture")
+	taskLogs.Flags().BoolVarP(&combined, "combined", "c", false, "dump all logs to one file")
+	taskLogs.Flags().StringVarP(&cwd, "cwd", "C", "", "change working directory for ouput")
+}
+
+func taskLogsMn(_ *cobra.Command, args []string) {
+	// Ensure project id exists
+	projectId := mustGetProjectID()
+
+	buildIdOrPackageName := args[0]
+	var buildId string
+
+	err := uuid.Validate(buildIdOrPackageName)
+	if err == nil {
+		buildId = buildIdOrPackageName
+	} else {
+		// argument is not a uuid, try to look up the most recent build for a package with said name
+		// projectCl := getClient(serviceProject).(peridotopenapi.ProjectServiceApi)
+		packageCl := getClient(servicePackage).(peridotopenapi.PackageServiceApi)
+		buildCl := getClient(serviceBuild).(peridotopenapi.BuildServiceApi)
+
+		_, _, err := packageCl.GetPackage(getContext(), projectId, "name", buildIdOrPackageName).Execute()
+		if err != nil {
+			errFatal(err)
+		}
+		// var pkg peridotopenapi.V1Package = *res.Package
+		// pkgId := pkg.GetId()
+
+		// try to get the latest builds for the package
+		res, _, err := buildCl.ListBuilds(
+			getContext(),
+			projectId).FiltersStatus(string(peridotopenapi.SUCCEEDED)).FiltersPackageName(buildIdOrPackageName).Execute()
+		if err != nil {
+			errFatal(err)
+		}
+
+		// TODO(neil): why is Total a string?
+		total, err := strconv.Atoi(*res.Total)
+		if err != nil {
+			errFatal(err)
+		}
+
+		// TODO(neil): support pagination
+		if total > int(*res.Size) {
+			panic("result set larger than one page")
+		}
+
+		if total > 0 {
+			builds := *res.Builds
+
+			// for _, build := range builds {
+			// 	buildjson, _ := build.MarshalJSON()
+			// 	log.Printf("build: %s", buildjson)
+			// }
+
+			// after sorting, the first build is the latest
+			buildId = builds[0].GetTaskId()
+		}
+	}
+
+	if cwd != "" {
+		os.Chdir(cwd)
```

We could want to fail with a clear error message if `Chdir` fails. Not silently continue writing files in an unexpected place or failing confusingly.